python/tskit/metadata.py

```diff
@@ -641,8 +638,8 @@ def __init__(self, schema: Mapping[str, Any] | None) -> None:
             except KeyError:
                 raise exceptions.MetadataSchemaValidationError(
                     f"Unrecognised metadata codec '{schema['codec']}'. "
-                    f"Valid options are {str(list(codec_registry.keys()))}."
-                )
+                    f"Valid options are {list(codec_registry.keys())!s}."
+                ) from None
```

Comment:
This `raise from None` thing is going to be a bit opaque to new people.

Reply:
I fail to see the value of that one too

Reply:
It makes the output  easier to read when exceptions occur, although developers may prefer to see the entire context chain of raised errors.